packages/dds/tree/src/test/simple-tree/treeApi.spec.ts

```diff
@@ -597,52 +590,155 @@ describe("treeNodeApi", () => {
+			const sb = new SchemaFactory("array-nodes-different-depths");
+			const list = sb.array("list", sb.number);
+			const treeSchema = sb.object("root", { list1: list, deeper: sb.object("inner", { list2: list})});
+
+			const root = hydrate(treeSchema, { list1: [1], deeper: { list2: [] }});
+
+			let list1ShallowChanges = 0;
+			let list1DeepChanges = 0;
+			let list2ShallowChanges = 0;
+			let list2DeepChanges = 0;
+			Tree.on(root.list1, "nodeChanged", () => {
+				assert.equal(list1ShallowChanges, 0, "list1 shouldn't have fire nodeChanged yet");
+				assert.equal(list1DeepChanges, 0, "list1 should not have fired treeChanged yet");
+				assert.equal(list2ShallowChanges, 0, "list2 shouldn't have fire nodeChanged yet");
+				assert.equal(list2DeepChanges, 0, "list2 should not have fired treeChanged yet");
+				list1ShallowChanges++;
+			});
+			Tree.on(root.list1, "treeChanged", () => {
+				assert.equal(list1ShallowChanges, 1, "list1 should have fire nodeChanged already");
+				assert.equal(list1DeepChanges, 0, "list1 should not have fired treeChanged yet");
+				assert.equal(list2ShallowChanges, 0, "list2 shouldn't have fire nodeChanged yet");
+				assert.equal(list2DeepChanges, 0, "list2 should not have fired treeChanged yet");
+				list1DeepChanges++;
+			});
+			Tree.on(root.deeper.list2, "nodeChanged", () => {
+				assert.equal(list1ShallowChanges, 1, "list1 should have fire nodeChanged already");
+				assert.equal(list1DeepChanges, 1, "list1 should have fired treeChanged already");
+				assert.equal(list2ShallowChanges, 0, "list2 shouldn't have fire nodeChanged yet");
+				assert.equal(list2DeepChanges, 0, "list2 should not have fired treeChanged yet");
+				list2ShallowChanges++;
+			});
+			Tree.on(root.deeper.list2, "treeChanged", () => {
+				assert.equal(list1ShallowChanges, 1, "list1 should have fire nodeChanged already");
+				assert.equal(list1DeepChanges, 1, "list1 should have fired treeChanged already");
+				// Test fails the next assert. I think it's because the delta visit still traverses the tree up to the
+				// array node that serves as target for the move, as it exitNode()'s back to the root, that currently enqueues
+				// subtreeChangedBatched event in anchorSet.ts, and only during the second pass of the delta visit we eventually
+				// enqueue the childrenChangedBatched event in anchorSet.ts, so we end up emitting a treeChanged event for the
+				// move-target node before we emit a nodeChanged event for it. This might be fixable by preprocessing the buffered
+				// events before emitting them but couldn't immediately come up with a solution. Discarding all subtreeChangedBatched
+				// events until we see a childrenChangedBatched for the same node doesn't work, because nodes that didn't have
+				// their own fields changed but are parents of other nodes which did, legitimately will only ever emit subtreeChangedBatched.
+				assert.equal(list2ShallowChanges, 1, "list2 should have fire nodeChanged already");
```

Comment:
This is how the list of bufferedEvents looks like 
![image](https://github.com/microsoft/FluidFramework/assets/716334/29b0407b-0970-49e9-9c71-5912b3d28cd0)

Everything starts fine with the chlidrenChangedBatched for the source array, subtreeChangedBatched events from there to the root, but then a subtreeChangedBatched on the node with `parentField: list2`, which is put in the list during the detach pass after the delta visit went to the target array node and did nothing with it.